### ISSUE TITLE
All new database queries are in `waiting` state

### ISSUE THREAD

**Issue by sebbacon** [ebmdatalab/openprescribing#157]
Example:

```
prescribing=> select query_start, waiting, state, substr(query, 0, 75) from pg_stat_activity;

          query_start          | waiting | state  |                                   substr                                   
-------------------------------+---------+--------+----------------------------------------------------------------------------
 2016-08-19 10:07:16.624689+00 | f       | active | SELECT SUM(p.quantity * r.percent_of_adq) AS numerator  FROM frontend_pres
 2016-08-19 08:07:53.457397+00 | t       | active | ALTER TABLE "frontend_orgbookmark" ADD CONSTRAINT "frontend_orgbookmark_pc
 2016-08-19 08:09:23.117937+00 | t       | active | SELECT "frontend_pct"."code", "frontend_pct"."ons_code", "frontend_pct"."n
 2016-08-19 08:10:50.824908+00 | t       | active | SELECT "frontend_pct"."code", "frontend_pct"."ons_code", "frontend_pct"."n
 2016-08-19 08:11:23.098649+00 | t       | active | SELECT "frontend_pct"."code", "frontend_pct"."ons_code", "frontend_pct"."n
 2016-08-19 08:11:55.222627+00 | t       | active | SELECT "frontend_pct"."code", "frontend_pct"."ons_code", "frontend_pct"."n
 2016-08-19 08:12:27.391626+00 | t       | active | SELECT "frontend_pct"."code", "frontend_pct"."ons_code", "frontend_pct"."n


```

**Comment by sebbacon** [comment]
All these processes were blocked on a migration completing.

I identified the process using this SQL from [here]():

```sql
  SELECT blocked_locks.pid     AS blocked_pid,
         blocked_activity.usename  AS blocked_user,
         blocking_locks.pid     AS blocking_pid,
         blocking_activity.usename AS blocking_user,
         blocked_activity.query    AS blocked_statement,
         blocking_activity.query   AS current_statement_in_blocking_process
   FROM  pg_catalog.pg_locks         blocked_locks
    JOIN pg_catalog.pg_stat_activity blocked_activity  ON blocked_activity.pid = blocked_locks.pid
    JOIN pg_catalog.pg_locks         blocking_locks 
        ON blocking_locks.locktype = blocked_locks.locktype
        AND blocking_locks.DATABASE IS NOT DISTINCT FROM blocked_locks.DATABASE
        AND blocking_locks.relation IS NOT DISTINCT FROM blocked_locks.relation
        AND blocking_locks.page IS NOT DISTINCT FROM blocked_locks.page
        AND blocking_locks.tuple IS NOT DISTINCT FROM blocked_locks.tuple
        AND blocking_locks.virtualxid IS NOT DISTINCT FROM blocked_locks.virtualxid
        AND blocking_locks.transactionid IS NOT DISTINCT FROM blocked_locks.transactionid
        AND blocking_locks.classid IS NOT DISTINCT FROM blocked_locks.classid
        AND blocking_locks.objid IS NOT DISTINCT FROM blocked_locks.objid
        AND blocking_locks.objsubid IS NOT DISTINCT FROM blocked_locks.objsubid
        AND blocking_locks.pid != blocked_locks.pid
 
    JOIN pg_catalog.pg_stat_activity blocking_activity ON blocking_activity.pid = blocking_locks.pid
   WHERE NOT blocked_locks.GRANTED;
```

... and killed it with `SELECT pg_cancel_backend(3505);`.

I don't yet understand why the migration (which was for staging) didn't complete.

**Comment by sebbacon** [comment]
The specific query that was blocking was

```sql
ALTER TABLE "frontend_orgbookmark" ADD CONSTRAINT "frontend_orgbookmark_pct_id_adec10d8_fk_frontend_pct_code" FOREIGN KEY ("pct_id") REFERENCES "frontend_pct" ("code") DEFERRABLE INITIALLY DEFERRED
```

That's just part of the SQL which creates a new table in a single transaction. There are no other locks when the transaction starts; I think this must be related to the very slow database updates mentioned in #148 and which are still ongoing: the CPU is still at 100%, and measures are still being calculated.

Closing this issue and continuing to track the performance in #148
